### PR TITLE
Lab Api Update

### PR DESCRIPTION
**Pull request by trwalke**
Migration to the new lab api

### CODE REVIEW

File: tests/Microsoft.Identity.Test.LabInfrastructure/LabServiceApi.cs [hunk 1]
```diff
@@ -60,90 +66,93 @@ private async Task<LabResponse> GetLabResponseFromApiAsync(UserQuery query)
...
-            {
-                user.InitializeHomeUser();
-            }
+            //if (!string.IsNullOrEmpty(response.User.HomeTenantId) && !string.IsNullOrEmpty(response.User.HomeUpn))
+            //{
+            //    response.User.InitializeHomeUser();
+            //}
 
-            return response;
+            return null;
```
Comment (bgavrilMS) [inline tests/Microsoft.Identity.Test.LabInfrastructure/LabServiceApi.cs]: Dead code and you seem to be always returning null.

File: tests/Microsoft.Identity.Test.LabInfrastructure/LabServiceApi.cs [hunk 2]
```diff
@@ -60,90 +66,93 @@ private async Task<LabResponse> GetLabResponseFromApiAsync(UserQuery query)
...
+        }
+
+        private static string StripBracketsFromLabResponse(string response)
```
Comment (bgavrilMS) [inline tests/Microsoft.Identity.Test.LabInfrastructure/LabServiceApi.cs]: Isn't the lab response a valid json?

File: tests/Microsoft.Identity.Test.LabInfrastructure/LabServiceApi.cs [hunk 3]
```diff
@@ -60,90 +66,93 @@ private async Task<LabResponse> GetLabResponseFromApiAsync(UserQuery query)
...
-
-            if (query.UserType != null)
+            if (query.FederationProvider != null)
             {
-                queryDict.Add(LabApiConstants.UserType, query.UserType.ToString());
+                queryDict.Add(LabApiConstants.FederationProvider, query.FederationProvider.ToString());
             }
 
-            queryDict.Add(LabApiConstants.External, query.IsExternalUser != null && (bool)(query.IsExternalUser) ? LabApiConstants.True : LabApiConstants.False);
+            //if (!string.IsNullOrWhiteSpace(query.Upn))
```
Comment (bgavrilMS) [inline tests/Microsoft.Identity.Test.LabInfrastructure/LabServiceApi.cs]: deal code?

File: tests/Microsoft.Identity.Test.LabInfrastructure/LabServiceApi.cs [hunk 4]
```diff
@@ -60,90 +66,93 @@ private async Task<LabResponse> GetLabResponseFromApiAsync(UserQuery query)
...
             {
-                queryDict.Add(LabApiConstants.B2CProvider, LabApiConstants.B2CFacebook);
+                queryDict.Add(LabApiConstants.SignInAudience, query.SignInAudience.ToString());
             }
 
-            if (query.B2CIdentityProvider == B2CIdentityProvider.Google)
-            {
-                queryDict.Add(LabApiConstants.B2CProvider, LabApiConstants.B2CGoogle);
-            }
+            //if (query.Licenses != null && query.Licenses.Count > 0)
```
Comment (bgavrilMS) [inline tests/Microsoft.Identity.Test.LabInfrastructure/LabServiceApi.cs]: deadl code

File: tests/Microsoft.Identity.Test.LabInfrastructure/LabServiceParameters.cs [hunk 5]
```diff
@@ -0,0 +1,94 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Microsoft.Identity.Test.LabInfrastructure
```
Comment (bgavrilMS) [inline tests/Microsoft.Identity.Test.LabInfrastructure/LabServiceParameters.cs]: Could we get comments on some of these? Some of the values, like `AutoMfaOnAll`, most of the ProtectionPolicy values  are not familiar.

File: tests/Microsoft.Identity.Test.LabInfrastructure/KeyVaultSecretsProvider.cs [hunk 6]
```diff
@@ -17,20 +17,14 @@ namespace Microsoft.Identity.Test.LabInfrastructure
...
     public class KeyVaultSecretsProvider : IDisposable
     {
-        private const string KeyVaultConfidentialClientId = "16dab2ba-145d-4b1b-8569-bf4b9aed4dc8";
-        private const string KeyVaultPublicClientId = "3c1e0e0d-b742-45ba-a35e-01c664e14b16";
-        private const string KeyVaultThumbPrint = "79FBCBEB5CD28994E50DAFF8035BACF764B14306";
-        private const string DataFileName = "data.txt";
-        private const string KeyVaultName = "https://buildautomation.vault.azure.net/";
-        private readonly KeyVaultConfiguration _config;
+        private const string _buildAutomationKeyVaultName = "https://buildautomation.vault.azure.net/";
+        private const string _mSIDLabLabKeyVaultName = "https://msidlabs.vault.azure.net";
```
Comment (jennyf19) [inline tests/Microsoft.Identity.Test.LabInfrastructure/KeyVaultSecretsProvider.cs]: >_mSIDLabLabKeyVaultName [](start = 29, length = 23)

nit: could you rename this? like _msidLabKeyVaultName

File: tests/Microsoft.Identity.Test.LabInfrastructure/LabApiConstants.cs [hunk 7]
```diff
@@ -26,11 +26,20 @@ public class LabApiConstants
...
+        public const string HomeDomain = "homedomain";
+        public const string HomeUPN = "homeupn";
+        public const string FederationProviderV2 = "federationprovider";
+        public const string AzureEnvironment = "azureenvironment";
+        public const string SignInAudience = "SignInAudience";
+
         public const string True = "true";
         public const string False = "false";
 
         public const string BetaEndpoint = "http://api.msidlab.com/api/userbeta";
```
Comment (jennyf19) [inline tests/Microsoft.Identity.Test.LabInfrastructure/LabApiConstants.cs]: question: is the beta endpoint the same for the new api?

**Comment by jennyf19** [comment]
> using System;

license info

---
Refers to: tests/Microsoft.Identity.Test.LabInfrastructure/LabAuthenticationHelper.cs:1 in 5f4d4ea. [](commit_id = 5f4d4eab6aac1a2df1f8c8d3ab0ef1287c919151, deletion_comment = False)

File: tests/Microsoft.Identity.Test.LabInfrastructure/LabServiceApi.cs [hunk 8]
```diff
@@ -16,10 +17,15 @@ namespace Microsoft.Identity.Test.LabInfrastructure
...
     {
         private readonly HttpClient _httpClient;
+        private string _labAccessAppId;
+        private string _labAccessClientSecret;
 
         public LabServiceApi()
         {
             _httpClient = new HttpClient();
+            KeyVaultSecretsProvider _keyVaultSecretsProvider = new KeyVaultSecretsProvider();
+            _labAccessAppId = _keyVaultSecretsProvider.GetMsidLabSecret("LabVaultAppID").Value;
```
Comment (jennyf19) [inline tests/Microsoft.Identity.Test.LabInfrastructure/LabServiceApi.cs]: nit: can you add these as constants somewhere?

File: tests/Microsoft.Identity.Test.LabInfrastructure/LabUserHelper.cs [hunk 9]
```diff
@@ -75,15 +75,14 @@ public static async Task<LabResponse> GetB2CMSAAccountAsync()
         public static Task<LabResponse> GetSpecificUserAsync(string upn)
         {
             var query = new UserQuery();
-            query.Upn = upn;
+            //query.Upn = upn;
```
Comment (jennyf19) [inline tests/Microsoft.Identity.Test.LabInfrastructure/LabUserHelper.cs]: remove?

File: tests/Microsoft.Identity.Test.LabInfrastructure/LabServiceParameters.cs (hunk 5 above)
Comment (jennyf19) [reply tests/Microsoft.Identity.Test.LabInfrastructure/LabServiceParameters.cs]: also, is there a reason the names don't follow standard practice (ex. camel casing)?

---
In reply to: [328545360](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1400#discussion_r328545360) [](ancestors = 328545360)

File: tests/Microsoft.Identity.Test.LabInfrastructure/UserType.cs [hunk 10]
```diff
@@ -3,10 +3,5 @@
 
 namespace Microsoft.Identity.Test.LabInfrastructure
 {
```
Comment (jennyf19) [inline tests/Microsoft.Identity.Test.LabInfrastructure/UserType.cs]: can this be deleted?

File: tests/Microsoft.Identity.Test.LabInfrastructure/LabAuthenticationHelper.cs [hunk 11]
```diff
@@ -0,0 +1,141 @@
...
+        private const string DataFileName = "data.txt";
+        private static LabAccessAuthenticationType s_defaultAuthType = LabAccessAuthenticationType.ClientCertificate;
+        private static string s_secret;
+
+        static LabAuthenticationHelper()
+        {
+            //The data.txt is a place holder for the keyvault secret. It will only be written to during build time when testing appcenter.
+            //After the tests are finished in appcenter, the file will be deleted from the appcenter servers.
+            //The file will then be deleted locally Via VSTS task.
+            if (File.Exists(DataFileName))
```
Comment (bgavrilMS) [inline tests/Microsoft.Identity.Test.LabInfrastructure/LabAuthenticationHelper.cs]: Do we still use this trick? I think we do not?

**Comment by bgavrilMS** [comment]
Will merge manually

**Pull request closed without merging**